packages/nimble-components/src/table/index.ts

```diff
@@ -1085,6 +1253,22 @@ export class Table<
 
         return tanstackSelectionState;
     }
+
+    private updateExpandedStateToDefault(): void {
+        const expandedState: { [key: string]: boolean } = {};
```

Comment:
Perhaps this can be called something better. The intent of this method is to collapse the expanded state to just that of the current available groupRows. So, there's nothing to remove, as all we're doing is adding to a default empty state.

Reply:
I think there may have been confusion about my question and where the line of code I chose to add it on. My question isn't about whether this function needed to be doing more. My question is about whether we needed to be doing more processing of the expanded state when data is updated to make sure the expanded state isn't trying to claim that removed rows are expanded.

Reply:
This logic has been updated and now is managed by an `ExpansionManager` class.